Sources/XToolSupport/DevCommand.swift

```diff
@@ -64,15 +64,23 @@ struct PackOperation {
         )
         let bundle = try await packer.pack()
 
-        if let entitlementsPath = plan.entitlementsPath {
-            let data = try await Data(reading: URL(fileURLWithPath: entitlementsPath))
-            let decoder = PropertyListDecoder()
-            let entitlements = try decoder.decode(Entitlements.self, from: data)
+        var allEntitlements = [URL: Entitlements]()
+
+        for product in plan.allProducts {
+            if let entitlementsPath = product.entitlementsPath {
+                let data = try await Data(reading: URL(fileURLWithPath: entitlementsPath))
+                let decoder = PropertyListDecoder()
+                let entitlements = try decoder.decode(Entitlements.self, from: data)
+                allEntitlements[product.resolveDir(bundle)] = entitlements
+            }
+        }
+
+        if !allEntitlements.isEmpty {
             print("Pseudo-signing...")
             try await Signer.first().sign(
                 app: bundle,
                 identity: .adhoc,
-                entitlementMapping: [bundle: entitlements],
+                entitlementMapping: allEntitlements,
```

Comment:
want to note that the zsign-backed `Signer` implementation doesn't handle mappings correctly right now (it signs all executables with the root app entitlements). The fix will probably be to explicitly invoke zsign inside-out, first signing the plugins and then the main bundle. But I don't think that needs to be under the purview of this PR, I can look into it in a followup.

Reply:
Thanks for letting me know. I had not tested entitlements for App Extensions but I believe this should be resolved in another pull request.

Reply:
For reference, this function should allow signing all entitlements? https://github.com/xtool-org/zsign/blob/fb10d47dd24bb84886532497bac149232e67f07a/Sources/Zupersign/signer.cpp#L67-L71